### PR TITLE
Fix Next page pagination x extraction path

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: custom-recipes/api-connect/recipe.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
         "icon": "icon-rocket"
     },
     "kind": "PYTHON",
+    "selectableFromDataset": "input_A_role",
     "inputRoles": [
         {
             "name": "input_A_role",
```
Comment (JaneBellaiche) [inline custom-recipes/api-connect/recipe.json:11]: could you change the "name" variable to something more meaningful?

File: python-lib/pagination.py [hunk 2]
```diff
@@ -66,7 +67,11 @@ def update_next_page(self, data, response_links=None):
                 self.is_last_batch_empty = True
             return
         elif self.counting_key:
-            batch_size = len(data.get(self.counting_key, []))
+            extracted_data = get_value_from_path(data, self.counting_key.split("."), can_raise=False)
+            if extracted_data:
+                batch_size = len(get_value_from_path(data, self.counting_key.split("."), can_raise=False))
```
Comment (JaneBellaiche) [inline python-lib/pagination.py:72]:
```suggestion
                batch_size = len(extracted_data)
```

File: python-lib/pagination.py [hunk 3]
```diff
@@ -66,7 +67,11 @@ def update_next_page(self, data, response_links=None):
...
+            else:
+                batch_size = 0
             if batch_size == 0:
                 self.is_last_batch_empty = True
```
Comment (JaneBellaiche) [inline python-lib/pagination.py:76]: this if is not needed `if batch_size == 0`: if extracted_data is True,` len(extracted_data)`>0, else, `batch_size = 0 `and `self.is_last_batch_empty` can be set to True
```suggestion
                batch_size = 0
                self.is_last_batch_empty = True
```

**Review by JaneBellaiche — CHANGES_REQUESTED** [review]
Bug fixed ! 
Another comment tho : a ValueError is raised if the extraction path is not found, but not if the Key to Next request URL is not found, is that expected ? Someone could have made a typo and if the first page is very big, he won't be able to see his mistake before really look into the output dataset

File: custom-recipes/api-connect/recipe.json (hunk 1 above)
Comment (alexbourret) [reply custom-recipes/api-connect/recipe.json:11]: I'd like to, but I've run some tests and changing this corrupts existing flows...

File: python-lib/pagination.py (hunk 2 above)
Comment (alexbourret) [reply python-lib/pagination.py]: Oops...
Done in a973c70

File: python-lib/pagination.py (hunk 3 above)
Comment (alexbourret) [reply python-lib/pagination.py]: Fixed in a973c70

**Comment by alexbourret** [comment]
@JaneBellaiche regarding the lack of error raised when Key to Next request URL is not found: it's a tricky one, because we don't know for sure how the targeted API reacts when there is no more page to be loaded. Some might keep the key and leave it empty, others will drop the key. And I can't raise on first page only: the targeted API might report a variable amount of data, which means extra page on day 1 but not an day 2...

**Review by JaneBellaiche — APPROVED** [review]
LGTM !

**Pull request merged**